src/views.jl

```diff
@@ -110,8 +110,30 @@ to_index(::BlockRange) = throw(ArgumentError("BlockRange must be converted by to
     to_indices(A, axes(A), I)
 
 
+# https://github.com/JuliaLang/julia/pull/30789
+if VERSION >= v"1.2-"
+
 # BlockSlices map the blocks and the indices
 # this is loosely based on Slice reindex in subarray.jl
+reindex(idxs::Tuple{BlockSlice{<:BlockRange}, Vararg{Any}},
+        subidxs::Tuple{BlockSlice{<:BlockRange}, Vararg{Any}}) =
+    (@_propagate_inbounds_meta; (BlockSlice(BlockRange(idxs[1].block.indices[1][Int.(subidxs[1].block)]),
+                                            idxs[1].indices[subidxs[1].indices]),
+                                 reindex(tail(idxs), tail(subidxs))...))
+
+reindex(idxs::Tuple{BlockSlice{BlockRange{1,Tuple{UnitRange{Int}}}}, Vararg{Any}},
+        subidxs::Tuple{BlockSlice{Block{1,Int}}, Vararg{Any}}) =
+    (@_propagate_inbounds_meta; (BlockSlice(Block(idxs[1].block.indices[1][Int(subidxs[1].block)]),
+                                            idxs[1].indices[subidxs[1].indices]),
+                                 reindex(tail(idxs), tail(subidxs))...))
+
+function reindex(idxs::Tuple{BlockSlice{Block{1,Int}}, Vararg{Any}},
+        subidxs::Tuple{BlockSlice{Block{1,Int}}, Vararg{Any}})
+    (idxs[1], reindex(tail(idxs), tail(subidxs))...)
```

Comment:
The corresponding 3-arg `reindex` has the check for `subidxs[1].block == Block(1)`.  Do we need to do this here as well?  Is it handled somewhere else already?

https://github.com/JuliaArrays/BlockArrays.jl/blob/f2f99cf996184327558e579f76db5b220c27460b/src/views.jl#L127-L131